src/sssom/writers.py

```diff
@@ -204,6 +219,31 @@ def write_owl(
 # Converters convert a mappingsetdataframe to an object of the supportes types (json, pandas dataframe)
 
 
+def _hydrate_axioms(
+    graph: rdflib.Graph,
+    *,
+    add_negative: bool = True,
+    add_no_term_found: bool = True,
+) -> None:
+    for axiom in graph.subjects(RDF.type, OWL.Axiom):
+        for p in graph.objects(subject=axiom, predicate=OWL.annotatedProperty):
+            for s in graph.objects(subject=axiom, predicate=OWL.annotatedSource):
+                for o in graph.objects(subject=axiom, predicate=OWL.annotatedTarget):
+                    if not add_negative and _is_negated(graph, axiom):
+                        continue
+                    if not add_no_term_found and _is_no_term_found(s, o):
+                        continue
+                    graph.add((s, p, o))
+
+
+def _is_no_term_found(s: rdflib.Node, o: rdflib.Node) -> bool:
+    return s == NO_TERM_REF or o == NO_TERM_REF
+
+
+def _is_negated(graph: rdflib.Graph, axiom: rdflib.Node) -> bool:
+    return any(obj == LITERAL_NOT for obj in graph.objects(subject=axiom, predicate=NEGATED))
```

Comment:
This works for now, but my understanding of how the LinkML RDF “dumper” works is that, as soon as we update SSSOM-Py to use the (as-yet-unreleased) version of the SSSOM schema, the `Not` value of the `predicate_modifier` will automatically be rendered as a `https://w3id.org/sssom/NegatedPredicate` resource instead of a `Not` string literal (because in the future schema the `Not` value has a `meaning` property set to `sssom:NegatedMapping`).

So when we do update, this function will need to be amended accordingly.

I’d suggest to future-proof it right now by making it look for _either_ a `Not` string literal _or_ a `sssom:NegatedPredicate` resource.